uniffi_macros/src/export/metadata.rs

```diff
@@ -0,0 +1,110 @@
+use std::{env, io};
+
+use camino::{Utf8Path, Utf8PathBuf};
+use fs_err::{self as fs};
+use once_cell::sync::Lazy;
+use proc_macro2::{Span, TokenStream};
+use quote::quote;
+use serde::Serialize;
+use tempfile::NamedTempFile;
+use uniffi_meta::Type;
+
+use super::ExportItem;
+
+mod function;
+
+use self::function::gen_fn_metadata;
+
+static METADATA_DIR: Lazy<Utf8PathBuf> = Lazy::new(|| {
+    let manifest_dir = env::var("CARGO_MANIFEST_DIR").expect("CARGO_MANIFEST_DIR must be set");
+    let metadata_dir = Utf8Path::new(&manifest_dir)
+        .join(".uniffi")
+        .join("metadata");
+
+    fs::create_dir_all(&metadata_dir).unwrap();
+    metadata_dir
+});
+
+pub(super) fn gen_metadata(item: syn::Item, mod_path: &[String]) -> syn::Result<ExportItem> {
+    match item {
+        syn::Item::Fn(item) => gen_fn_metadata(item, mod_path),
+        syn::Item::Impl(_) => Err(syn::Error::new(
+            Span::call_site(),
+            "impl blocks are not yet supported",
+        )),
+        // FIXME: Support const / static?
```

Comment:
What do you think about supporting a module here as well?  That would be equivalent to wrapping each item in the module.

Some users like how the UDL lets you see the entire API in one place.  In a proc-macro world, I think that would translate to a single ffi module with the `uniffi:export` attribute.

Reply:
I guess it could work 🤷🏼

Doing it *well* (w.r.t. error handling) would be a decent amount of work though, I think.